contracts/validator-manager/interfaces/IValidatorManager.sol

```diff
@@ -0,0 +1,175 @@
+// (c) 2024, Ava Labs, Inc. All rights reserved.
+// See the file LICENSE for licensing terms.
+
+// SPDX-License-Identifier: Ecosystem
+
+pragma solidity 0.8.25;
+
+enum ValidatorStatus {
+    Unknown,
+    PendingAdded,
+    Active,
+    PendingRemoved,
+    Completed,
+    Invalidated
+}
+
+struct Validator {
+    ValidatorStatus status;
+    bytes32 nodeID;
+    uint64 startingWeight;
+    uint64 messageNonce;
+    uint64 weight;
+    uint64 startedAt;
+    uint64 endedAt;
+}
+
+struct ValidatorChurnPeriod {
+    uint256 startedAt;
+    uint256 initialWeight;
+    uint256 totalWeight; // TODO add initial validator set to total weight.
+    uint64 churnAmount;
+}
+
+/**
+ * @notice Event emitted when validator weight is updated.
+ * @param validationID The ID of the validation period
+ * @param nonce The message nonce used to update the validator weight
+ * @param validatorWeight The updated validator weight that is sent to the P-Chain
+ * @param setWeightMessageID The ID of the Warp message that updates the validator's weight on the P-Chain
+ */
+event ValidatorWeightUpdate(
```

Comment:
why do we have this event outside of interface definition? Usually it's been included inside the interface